File: FinSim.Core/Contributions/IRAContributionModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace FinSim.Core.Contributions
{
    public class IRAContributionModel : IContribution
    {
        private const decimal FedMaximum = 5000m;
        private readonly Frequency _frequency;

        public IRAContributionModel(Frequency frequency)
        {
            _frequency = frequency;
        }

        public decimal GetContributionAmount(int yearIndex, int weekIndex)
        {
            if (weekIndex % (int)_frequency != 0)
                return 0m;

            return FedMaximum / (decimal)(Constants.NumberOfWeeks / (int)_frequency);
        }
    }
}

File: FinSim/src/app/models/contributions/IRAContributionModel.ts
import { IContribution } from './contribution.interface';
import { Frequency } from '../frequency.enum';
import { NumberOfWeeks, ContributionTypes } from '../constants';
import { CurrencyPipe } from '@angular/common';

export class IRAContributionModel implements IContribution {

    public type = ContributionTypes.ContributionIRAType;

    public annualAmount = 0;
    public max = true;
    public frequency = Frequency.BiWeekly;

    private currency: CurrencyPipe = new CurrencyPipe('en-us');

    constructor(public investmentId: number,
                init?: Partial<IRAContributionModel>) {

        if (init) {
            Object.assign(this, init);
        }
    }

    public getContributionAmount(yearIndex: number, weekIndex: number): number {
        if (weekIndex % <number>this.frequency !== 0 ) {
            return 0;
        }

        if (this.max) {
            this.annualAmount = 5000;
        }

        return this.annualAmount / (NumberOfWeeks / <number>this.frequency);
    }

    public displayString() {
        if ( this.max ) {
            return `Maximum Annual IRA Contribution`;
        }

        return `${this.currency.transform(this.annualAmount, 'USD', 'symbol')} Annual IRA Contribution`;
    }
}
